scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -133,4 +229,29 @@ private void extractAnalyzerImportsAndDependencies(ScanTomlFile.Analyzer analyze
     private void buildStringWithNewLine(StringBuilder stringBuilder, String content) {
         stringBuilder.append(content).append(System.lineSeparator());
     }
+
+    List<Issue> runExternalAnalyzers(Project project, Map<String, List<Rule>> externalAnalyzers) {
+        List<ScannerContext> scannerContextList = new ArrayList<>();
```

Comment:
I think you can initialize this with initial capacity as well since you add one for each in externalAnalyzers map